libsrc/grabber/x11/X11Wrapper.cpp

```diff
@@ -23,13 +23,17 @@ void X11Wrapper::action()
 		{
 			stop();
 		}
+		else
+		{
+			if (_grabber.updateScreenDimensions() < 0 )
+			{
+				stop();
+			}
+		}
 	}
 
 	if (isActive())
 	{
-		if (_grabber.updateScreenDimensions() >= 0 )
```

Comment:
Is there a reason for calling updateScreenDimensions every time we want to grab a frame ?